### PR TITLE
Issue #22: build guess-pool and answer-pool policy engine

### LINKED ISSUE #22: Build guess-pool and answer-pool policy engine (base + irregular answer policy)
## Outcome
Separate guess validation pool from answer selection pool.

## Deliverables
- Guess pool includes valid expanded forms (post-pruning).
- Answer pool follows base + irregular policy.
- Policy versioning recorded in generated metadata.

## Dependencies (Updated)
- Requires: `#21`
- Feeds: `#23`, `#26`
- **Not blocked by full Epic #6**

## Acceptance Criteria
- Guess acceptance and answer selection behavior match policy definition.
- Regression tests cover plural/past/ing edge cases.
- Policy decisions are explicit and traceable in persisted metadata.

## Cross-link
Epic #17 MVP ownership.

### PR DESCRIPTION
**Pull request by curdriceaurora**
## Summary
- add `lib/provider-pool-policy.js` to derive deterministic provider gameplay pools
- generate `guess-pool.txt`, `answer-pool.txt`, and `pool-policy.json` from source-manifest + expanded forms
- implement base+irregular answer policy: `.dic` stems as base answers plus optional irregular allowlist entries present in guess pool
- document new artifacts and policy gotchas in provider import contract
- add focused tests covering deterministic output, plural/past/ing exclusion, missing inputs, and fail-closed empty-answer behavior

## Why
Issue #22 requires policy separation between permissive guesses and curated answers, with explicit persisted metadata for auditability and downstream integration in #23/#26.

## Validation
- `npm test -- tests/provider-pool-policy.test.js`
- `npm run check`
- `npm run test:all`

Closes #22

### CODE REVIEW

File: lib/provider-pool-policy.js [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+      "PERSISTENCE_WRITE_FAILED",
+      `Failed to persist file at ${filePath}.`,
+      { cause: err }
+    );
+  }
+}
+
+async function writeJsonAtomic(filePath, payload) {
+  await writeFileAtomic(filePath, `${JSON.stringify(payload, null, 2)}\n`);
+}
```
[the remark this answers is not in the document]
Comment (curdriceaurora) [reply lib/provider-pool-policy.js]: Addressed in fb1248e by extracting shared provider artifact utilities into `lib/provider-artifact-shared.js` and consuming them from both `lib/provider-hunspell.js` and `lib/provider-pool-policy.js` (commit/path normalization, variant allowlist source, and atomic write logic). Also added a guardrail check to prevent this drift from reappearing.
